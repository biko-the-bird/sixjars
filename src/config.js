
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

//this is a configuration for enzyme that needs
//to be imported in each testing file
const config = configure({ adapter: new Adapter() });
export default config;