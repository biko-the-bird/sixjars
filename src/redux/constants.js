export const SET_JARS = "SET_JARS";
export const WITHDRAW_JAR = "WITHDRAW_JAR";
export const JARS_COOKIE = "JARS_COOKIE";
export const NEC = "NEC";
export const FFA = "FFA";
export const EDU = "EDU";
export const GIV = "GIV";
export const LTS = "LTS";
export const PLA = "PLA";


export const INIT_JARS = {NEC: 0, FFA: 0,GIV: 0, PLA: 0,EDU: 0,LTS: 0};