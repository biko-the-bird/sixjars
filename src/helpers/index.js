import * as constants from '../redux/constants';


export const updateJars = ( deposit, jars) => {
    const keys = Object.keys(jars);
    keys.forEach(key => {
        switch(key) {
            case constants.NEC: 
                jars[key] = jars[key] + (deposit * 0.55);
                break;
            case constants.GIV: 
                jars[key] = jars[key] + (deposit * 0.05);
                break;
            default:
                jars[key] = jars[key] + (deposit * 0.1);
                break;
        }
    });
    return jars;
}