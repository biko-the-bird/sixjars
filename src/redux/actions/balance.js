
import * as constants from '../constants';

export const setJars = (jars) => {
    return {
        type: constants.SET_JARS,
        jars
    }
};

export const withdrawJar = (jar, amount) => {
    return {
        type: constants.WITHDRAW_JAR,
        jar,
        amount
    }
}

