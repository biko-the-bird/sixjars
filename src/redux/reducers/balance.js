import * as constants from '../constants';
import {read_cookie, bake_cookie} from 'sfcookies';




const balance = (prevState = constants.INIT_JARS, action) => {
    let balance;
    const oldJars = read_cookie(constants.JARS_COOKIE) || prevState;
    
    switch (action.type) {
        case constants.SET_JARS:
            balance = action.jars;
            break;
        case constants.WITHDRAW_JAR:
            balance = oldJars;
            var difference =  balance[action.jar] - action.amount;
            balance[action.jar] = (difference >= 0) ? difference : 0;
        default:
            balance = oldJars;
    }
    console.log("reducer balance", balance);
    bake_cookie(constants.JARS_COOKIE, balance);
    return balance;
}
export default balance;

