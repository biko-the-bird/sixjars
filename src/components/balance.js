import React, { Component } from "react";
import { connect } from "react-redux";
import { setJars, withdrawJar } from "../redux/actions/balance";
import { updateJars } from "../helpers";
import * as constants from '../redux/constants';

export class Balance extends Component {
  constructor(props) {
    super(props);
    this.state = {
      deposit: 0,
      withdraw: 0,
      withdrawJar: constants.NEC 
    };
    this.handleDeposit = this.handleDeposit.bind(this);
  }

  handleDeposit = () => {
    var jars = updateJars(this.state.deposit, this.props.jars);

    this.props.setJars(jars);
    this.setState({
      deposit: 0,
    });
  };

  handleWithdraw = () => {
    this.props.withdrawJar(this.state.withdrawJar, this.state.withdraw);
    this.setState({
        withdraw: 0,
    });
  }

  calcTotal = () => {
    var keys = Object.keys(this.props.jars);
    var t = 0;
    keys.forEach(key => {
      t += this.props.jars[key];
    });
    return t;
  };


  render() {
    const total = this.props.jars ? this.calcTotal() : 0;
    var keys = Object.keys(this.props.jars);
    return (
      <div>
        Total Balance: ${total.toFixed(2)}
        <div>
          <label>Deposit</label>
          <input
            type="number"
            className="deposit-input"
            value={this.state.deposit}
            onChange={e => this.setState({ deposit: e.target.value })}
          />
          <button className="deposit-btn" onClick={() => this.handleDeposit()}>
            Deposit
          </button>
        </div>
        <ul>
          {keys.map(k => {
            return (
              <li key={k}>
                {k}: ${this.props.jars[k].toFixed(2)}
              </li>
            );
          })}
        </ul>
        <hr/>
        <div>
<h3>Make a withdraw</h3>
        </div>
        <select id="jar-selector" onChange={e => this.setState({withdrawJar: e.target.value})}>
          {keys.map(key => {
            return (
              <option value={key} key={key}>
                {key} 
              </option>
            );
          })}
        </select>
        
        <br/>
        <input
            type="number"
            className="withdraw-input"
            value={this.state.withdraw}
            onChange={e => this.setState({withdraw: e.target.value})}
          />
          <button className="withdraw-btn" onClick={() => this.handleWithdraw()}>
            Withdraw
          </button>
        
          <br/>
        <br/>
       
      </div>
    );
  }
}

const mapStateToProps = state => {
  return { jars: state };
};
const mapActions = {
  setJars,
  withdrawJar
};
export default connect(
  mapStateToProps,
  mapActions
)(Balance);
