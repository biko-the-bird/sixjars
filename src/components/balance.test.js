import React from 'react';
import config from '../config';
import {shallow} from 'enzyme';
import {Balance} from './balance';
import {updateJars} from '../helpers';
import * as constants from '../redux/constants';


describe("balance", () => {
    const mockSetJars = jest.fn();
    const mockWithdrawJar = jest.fn();
    const balance = shallow(<Balance setJars={mockSetJars} jars={constants.INIT_JARS} withdrawJar={mockWithdrawJar}/>);
    it("renders", () => {
        expect(balance).toMatchSnapshot();
    })
    

   it("contains a card component", () => {
    expect(balance.find(".balance-card").exists()).toBe(true);
   });



    
})