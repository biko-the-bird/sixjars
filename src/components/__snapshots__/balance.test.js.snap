// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`balance renders 1`] = `ShallowWrapper {}`;
